app/models/user.rb

```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
   # Existing associations
+  has_many :libraries, dependent: :destroy
+  has_many :blogs, dependent: :destroy
+  has_many :gift_cards, dependent: :destroy
```

Comment:
The 'email' field in the User model is validated for presence and uniqueness but also allows blank values due to 'allow_blank: true'. This is contradictory and could lead to unexpected behavior. Remove 'allow_blank: true' if the intention is to require an email address.